utils/crx-webpack-plugin/index.js

```diff
@@ -0,0 +1,86 @@
+var fs = require('fs');
+var path = require('path');
+var join = path.join;
+var mkdirp = require('mkdirp');
+var ChromeExtension = require('crx');
+
+function CrxWebpackPlugin(options) {
+  this.options = options || {};
+  if (!this.options.updateUrl) {
+    this.options.updateUrl = 'http://localhost:8000/';
+  }
+  if (!this.options.updateFilename) {
+    this.options.updateFilename = 'updates.xml';
+  }
+
+  // remove trailing slash
+  this.options.updateUrl = this.options.updateUrl.replace(/\/$/, '');
+
+  // setup paths
+  this.context = path.dirname(module.parent.filename);
+  this.keyFile = path.isAbsolute(this.options.keyFile)
+    ? this.options.keyFile
+    : join(this.context, this.options.keyFile);
+  this.outputPath = path.isAbsolute(this.options.outputPath)
+    ? this.options.outputPath
+    : join(this.context, this.options.outputPath);
+  this.contentPath = path.isAbsolute(this.options.contentPath)
+    ? this.options.contentPath
+    : join(this.context, this.options.contentPath);
+
+  // set output info
+  this.crxName = this.options.name + '.crx';
+  this.crxFile = join(this.outputPath, this.crxName);
+  this.updateFile = join(this.outputPath, this.options.updateFilename);
+  this.updateUrl = this.options.updateUrl + '/' + this.options.updateFilename;
+
+  // initiate crx
+  this.crx = new ChromeExtension({
+    privateKey: fs.readFileSync(this.keyFile),
+    codebase: this.options.updateUrl + '/' + this.crxName,
+  });
+}
+
+// hook into webpack
+CrxWebpackPlugin.prototype.apply = function (compiler) {
+  var self = this;
+  self.logger = compiler.getInfrastructureLogger('crx-webpack-plugin');
+  return compiler.hooks.done.tap('crx-webpack-plugin', function () {
```

Comment:
crx-webpack-plugin does not support webpack5 currently because the author seems not active in this project for a long time. But there is just one API change required to make all things work again, so I modify the source code and put it into our utils dir.